third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc

```diff
@@ -344,37 +426,65 @@ function buildProtocolResult({ result, exceptionDetails }) {
   return { result: protocolResult };
 }
 
-function Pause_evaluateInFrame({ frameId, expression }) {
-  const frames = getStackFrames();
-  const index = +frameId;
-  assert(index < frames.length);
-  const frame = frames[index];
 
-  const rv = doEvaluation();
-  return buildProtocolResult(rv);
+function Pause_evaluateInFrame({ frameId, expression }) {
+  try {
+    const result = window.DevOnly?.tryEvalDev(expression, frameId);
+    if (result) {
+      return result;
+    }
 
-  function doEvaluation() {
-    // In order to do the evaluation in the right frame, the same number of
-    // frames need to be on V8's stack when we do the evaluation as when we got
-    // the stack frames in the first place. The debugger agent extracts a frame
-    // index from the ID it is given and uses that to walk the stack to the
-    // frame where it will do the evaluation (see DebugStackTraceIterator).
-    return sendMessage(
-      "Debugger.evaluateOnCallFrame",
-      {
-        callFrameId: frame.callFrameId,
-        expression,
-      }
-    );
+    const frames = getStackFrames();
+    const index = +frameId;
+    assert(index < frames.length);
+    const frame = frames[index];
+
+    const rv = doEvaluation();
+    return buildProtocolResult(rv);
+
+    function doEvaluation() {
+      // In order to do the evaluation in the right frame, the same number of
+      // frames need to be on V8's stack when we do the evaluation as when we got
+      // the stack frames in the first place. The debugger agent extracts a frame
+      // index from the ID it is given and uses that to walk the stack to the
+      // frame where it will do the evaluation (see DebugStackTraceIterator).
+      return sendMessage(
+        "Debugger.evaluateOnCallFrame",
+        {
+          callFrameId: frame.callFrameId,
+          expression,
+        }
+      );
+    }
+  }
+  catch (err) {
+    return { result: { data: {}, exception: { value: err.stack } } };
   }
 }
 
 function Pause_evaluateInGlobal({ expression }) {
-  const rv = sendMessage("Runtime.evaluate", { expression });
-  return buildProtocolResult(rv);
+  try {
+    const result = window.DevOnly?.tryEvalDev(expression);
+    if (result) {
+      return result;
+    }
+
+    const rv = sendMessage("Runtime.evaluate", { expression });
+    return buildProtocolResult(rv);
+  }
+  catch (err) {
+    return { result: { data: {}, exception: { value: err.stack } } };
+  }
 }
 
+// function onMaybeNewPause() {
+// }
+
 function Pause_getAllFrames() {
+  // // we don't currently have an event for `Pause.createPause`, so we use this instead.
+  // //  (→ this is called first from `Pause.createPause`, but also from other places)
+  // onMaybeNewPause();
```

Comment:
this commented out code should be removed

Reply:
* Removed [here](https://github.com/replayio/chromium/commit/195e8ff70f647e16d6eb8ff5e88b90388d1c24cb#diff-dfa4df8e69a4e28b9a02e4b123474269de882574bd059edad8100b43fcc073bbR435).
* The `"called first"` comment has also been removed (since it might end up getting outdated in the near future)